docs/UsersGuide/source/ContributorsGuide.rst

```diff
@@ -0,0 +1,359 @@
+
+
+.. _ContribStandards:
+
+Code and Configuration Standards
+===================================
+
+General
+-----------
+
+Platform-specific settings should be handled only through configuration and module files, not in code or scripts.
+
+SRW Application
+------------------
+
+Externals.cfg
+    * All externals live in a single ``Externals.cfg`` file.
+    * Externals should point only to authoritative repositories for the given code base.
+
+        * Temporary exceptions are made for a PR into the ``develop`` branch of ``ufs-srweather-app`` that is dependent on another PR (e.g., a ``regional_workflow`` PR from the same contributor). When the component PR is merged, the contributor must update the corresponding ``ufs-srweather-app`` PR with the hash of the component's authoritative repository.
+    * Only a single hash will be maintained for any given external code base. All externals should point to this static hash (not to the top of a branch). 
+    
+Build system
+    * Each component should build with CMake
+    * Each component should build with Intel compilers on official `Level 1 <https://github.com/ufs-community/ufs-srweather-app/wiki/Supported-Platforms-and-Compilers>`__ platforms and with GNU or Intel compilers on other platforms. 
+    * Each component should have a mechanism for platform independence (i.e., no hard-coded machine-specific settings outside of established environment, configuration, and module files). 
+    * Each component should build with the latest release of the `HPC-Stack <https://github.com/NOAA-EMC/hpc-stack>`__. 
+
+Module files (env files)
+    * Each component should build using the common modules located in the ``env/srw_common`` file.
+
+..
+    COMMENT: Common what?! Add update once this is clarified. 
+
+Regional Workflow
+----------------------
+The ``regional_workflow`` repository must not contain source code for compiled programs. Only scripts, configuration files, and documentation should reside in this repository. 
+
+**General Coding Standards:** 
+    * All bash scripts must explicitly be ``#!/bin/bash`` scripts. They should *not* be login-enabled.
+    * MacOS does not have all Linux utilities by default. Developers should ensure that they do not break any MacOS capabilities with their contribution.
+    * All code must be indented appropriately and conform to the style of existing scripts (e.g., local variables should be lowercase, global variables should be uppercase).
+
+..
+    COMMENT: And what is this style? Do we have a style guide? Ask about linking to this `style guide <https://google.github.io/styleguide/shellguide.html>`__
+
+**Python Coding Standards:** 
+    * All Python code contributions should come with an appropriate ``environment.yml`` file for the feature. 
```

Comment:
should this be environment.yaml?

Reply:
It's copied from @mkavulich 's code management doc. I've seen both. A brief internet search suggests that they are synonymous, but now that 3-letter file extensions are no longer a requirement on most systems, .yaml is more common. I'll change it to .yaml unless I hear (soon) from anyone that there's some reason to keep .yml